src/coreclr/vm/i386/cgenx86.cpp

```diff
@@ -200,6 +200,41 @@ void EHContext::UpdateFrame(PREGDISPLAY regs)
 }
 #endif // FEATURE_EH_FUNCLETS
 
+
+void UnmanagedToManagedFrame::UpdateRegDisplay(const PREGDISPLAY pRD)
+{
+    CONTRACT_VOID
+    {
+        NOTHROW;
+        GC_NOTRIGGER;
+        MODE_ANY;
+        HOST_NOCALLS;
+        SUPPORTS_DAC;
+    }
+    CONTRACT_END;
+
+    pRD->PCTAddr = GetReturnAddressPtr();
+#define CALLEE_SAVED_REGISTER(regname) pRD->p##regname = (DWORD*) &m_calleeSavedRegisters.regname;
+    ENUM_CALLEE_SAVED_REGISTERS();
+#undef CALLEE_SAVED_REGISTER
+
+    pRD->ControlPC = *PTR_PCODE(pRD->PCTAddr);
+    UINT cbStackPop = GetMethod()->CbStackPop();
```

Comment:
If I am reading the code correctly, `GetMethod` is going to return the managed method that the UMThunk or ComMethodFrame is wrapping.

Is it the managed method to ask for `CbStackPop`? I suspect we need to `CbStackPop` for the unmanaged signature with unmanaged view for the arguments here.

Reply:
Hmm, yes, you are right, I have forgotten that the unmanaged view can be different.

Reply:
@janvorli Do we have a test for this? If you can offer a simple strawman example the Interop team can help with adding the appropriate testing assets.

Reply:
@AaronRobinsonMSFT to test this fully, we would need:
* A COM test that exposes a managed method via COM and calls it from other managed method
* That method would need to have arguments passed on stack
* The unmanaged signature would need to differ from the managed one in the size of arguments passed on stack.

It would be great if someone from the interop team can help me with creating that.

Reply:
@AaronRobinsonMSFT I also have a question about getting the unmanaged signature for COM method. Would the right one be the one returned by the `ComCallMethodDesc::GetCallMethodDesc`?